### PR TITLE
Hashtable: misc tweaks

### PR DESCRIPTION
**Pull request by cgzones**
- use more distinct typename for key type
- introduce typename for size type
- fail hard on too big size request
- hide implementation of Hashtable and HashtableItem

### CODE REVIEW

**Review by BenBE — COMMENTED** [review]
LGTM so far.

Any particular reason for not using `uint64_t` for `ht_size_t`?

**Comment by cgzones** [comment]
> Any particular reason for not using uint64_t for ht_size_t?

 I didn't want to risk a slow-down on 32bit architectures; switched to `size_t`.

File: Hashtable.c [hunk 1]
```diff
@@ -80,18 +94,16 @@ static const uint64_t OEISprimes[] = {
    34359738337, 68719476731, 137438953447
 };
 
-static uint64_t nextPrime(unsigned int n) {
-   assert(n <= OEISprimes[ARRAYSIZE(OEISprimes) - 1]);
```
Comment (BenBE) [inline Hashtable.c:84]: Why not keep this assert?

File: Hashtable.c [hunk 2]
```diff
@@ -194,15 +206,19 @@ void Hashtable_setSize(Hashtable* this, unsigned int size) {
...
+void Hashtable_put(Hashtable* this, ht_key_t key, void* value) {
 
    assert(Hashtable_isConsistent(this));
    assert(this->size > 0);
    assert(value);
 
    /* grow on load-factor > 0.7 */
-   if (10 * this->items > 7 * this->size)
+   if (10 * this->items > 7 * this->size) {
+      if (((ht_size_t)-1) / 2 < this->size)
```
Comment (BenBE) [inline Hashtable.c:217]: Actually would also need to check for last index of OEIS prime array.

**Review by BenBE — COMMENTED** [review]
Splitting pure type changes and actual code tweaks would be easier for review.

**Comment by cgzones** [comment]
Split the changes into multiple commits and reworked the logic for reaching to big primes.

**Review by BenBE — APPROVED** [review]
LGTM.

**Pull request merged**